internal/common/logs.go

```diff
@@ -0,0 +1,156 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"sync"
+	"time"
+)
+
+// StartLogWrite starts go routines to write logs to each of the logs that are monitored by CW Agent according to
+// the config provided
+func StartLogWrite(configFilePath string, agentRunDuration time.Duration, dataRate int) error {
+	//create wait group so main test thread waits for log writing to finish before stopping agent and collecting data
+	var wg sync.WaitGroup
+
+	logPaths, err := getLogFilePaths(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	for _, logPath := range logPaths {
+		filePath := logPath //necessary weird golang thing
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err = writeToLogs(filePath, agentRunDuration, dataRate)
```

Comment:
It looks like you have a single `err` variable, and multiple goroutines will set it.
Unpredictible return value.

Maybe use an error counter that sums errors from all the files.

Reply:
I use this in [metrics section but forgot to do that in logs](https://github.com/aws/amazon-cloudwatch-agent-test/pull/109/files#diff-781bb3810ede78771b4deb63bb6bb797a2d06434f0d844429b5ad16a20304e0fR19-R43).